tests/run-unit-tests.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+#
+# Unit-testing script
+#
+# This script does the following:
+#   1. Check whether any files were modified that would necessitate unit testing (using the `TRAVIS_COMMIT_RANGE` environment variable).
+#   2. If there are no changed files that require unit testing, exit successfully.
+#   3. Otherwise, run all the unit tests.
+#
+# We do things this way because our unit testing takes a rather long
+# time (average 18-19 minutes as of the original creation of this script),
+# so skipping it when we don't actually need it can significantly speed
+# up the CI process.
+#
+# Copyright: SPDX-License-Identifier: GPL-3.0-or-later
+#
+# Author: Austin S. Hemmelgarn <austin@netdata.cloud>
+#
+# shellcheck disable=SC2230
+
+install_netdata() {
+  echo "Installing Netdata"
+  fakeroot ./netdata-installer.sh \
+    --install "$HOME" \
+    --dont-wait \
+    --dont-start-it \
+    --enable-plugin-nfacct \
+    --enable-plugin-freeipmi \
+    --disable-lto
+}
+
+c_unit_tests() {
+  echo "Running C code unit tests"
+  "$HOME"/netdata/usr/sbin/netdata -W unittest
+}
+
+install_netdata || exit 1
+
+c_unit_tests || exit 1
```

Comment:
This script can probably be folded directly into the GHA workflow. The primary reason it was a script at all was so that I could do the commit checking in travis to skip the unit tests if nothing that they tested was changed.

Reply:
Agreed, but let's do it later :) No need to block this PR; we can just put up another PR to fold this later when we have more time!